route/teams.js

```diff
@@ -0,0 +1,233 @@
+  }).sort(byName);
+
+  const weeks = team.schedule.map(w => {
+    const isHome = w.side == 'vs';
+    const match = matches.get(w.match_key);
+    let result = null;
+    if(match) {
+      const points = match.getPoints();
+      //result -> '(W|L) team.score - opp.score'
+      if(match.isDone()) {
+        result = isHome ?
+          ((points.home > points.away) ? 'W' : 'L') :
+          ((points.away > points.home) ? 'W' : 'L');
+      }
+      else {
+        result = 'R' + match.round;
+      }
+      result += ' ';
+      const ts = isHome ? points.home : points.away;
+      const os = isHome ? points.away : points.home;
+      result += ts + '-' +os;
+    }
+
+    const [year, month, day] = [
+      w.date.slice(0, 4),
+      w.date.slice(4, 6),
+      w.date.slice(6, 8),
+    ];
+
+    return {
+      week: w.week,
+      date: `${month}-${day}-${year}`,
+      side: w.side,
+      opp:  w.opp,
+      key: w.match_key,
+      result: result
+    };
+  });
+
+  const html = mustache.render(base,{
+    canAdd: req.user.isLeagueAdmin,
+    canRemove: req.user.isLeagueAdmin,
+    team_id: team.key,
+    title: team.name,
+    name: team.name,
+    division: team.division,
+    venue: vname,
+    captain: team.captain,
+    co_captain: team.co_captain,
+    team_rating: teamRating,
+    roster: lineup,
+    schedule: weeks,
+    sugs: JSON.stringify(getSuggestions(), null, 2),
+  }, {
+    content: template
+  });
+
+  res.send(html);
+});
+
+router.post('/teams/:team_id/roster/add', function(req,res) {
+  // TODO: Who is trying to do this, and are they allowed?
+
+  const team = getTeam(req.params.team_id);
+  // TODO: It might be better to return a 500 or something, but since we
+  //  are returning rendered html, that's not as good a model.
+  if(!team) { return res.redirect('/teams'); }
+
+  if(!req.body || !req.body.name || !req.body.role) {
+    // TODO: redirect to /teams/:team_id, but with an error msg.
+    console.log('Jacked up or missing body');
+  }
+
+  // TODO: Do we need to lookup the player in some way?
+  //  If the player wasn't in the suggestions, it may be
+  //  difficult to validate anyways.
+
+  const { name, role } = req.body;
+
+  // TODO: What happens if role is already taken?
+
+  switch(role) {
+    case 'P':
+      // No need to do anything special.
+    case 'A':
+      // TODO: Is these supposed to be a player object, key, name? Check the json
+      team.co_captain = name;
+    case 'C':
+      team.captain = name;
+  }
+
+  // Is the player already on the team?
+  if(!team.roster.find(p => p.name === name)) {
+    // TODO: We likely do need some kind of object shape here.
+    team.roster.push({ name });
+  }
+
+  // TODO: save changes
+
+  // TODO: Redirect with a success msg (but first need to support messages)
+  res.redirect(`/teams/${team.key}`);
+});
+
+router.post('/teams/:team_id/roster/remove', function(req,res) {
+  console.log('POST remove', req.params, req.body);
+  // TODO: Check permissions
```

Comment:
We might be able to get away without checking permissions at first because you will not see the controls to POST changes. Though that would not prevent a smart enough hacker from being able to make unauthorized posts. I'm leaning towards adding the checks before going live.